pkg/castemplate/v1alpha1/cas_template_engine.go

```diff
@@ -122,8 +122,8 @@ func initValues() (v map[string]interface{}) {
 	}
 }
 
-// New returns a new instance of engine
-func New(cast *v1alpha1.CASTemplate, key string, values map[string]interface{}) (e *engine, err error) {
+// NewEngine returns a new instance of engine of version v1alpha1
+func NewEngine(cast *v1alpha1.CASTemplate, key string, values map[string]interface{}) (e *engine, err error) {
```

Comment:
This can be `New(..)` without suffixing `Engine`.
We will suffix only when there are multiple `New` coming up.

Reply:
changed it to `Engine(...)`  as suggested by you `cast.Engine()` is more readable.